apollo-router/src/services/layers/persisted_queries/mod.rs

```diff
@@ -84,7 +80,15 @@ impl PersistedQueryLayer {
                 if let Some(previous_manifest_poller) = preexisting_manifest_poller.clone() {
                     previous_manifest_poller
                 } else {
-                    Arc::new(PersistedQueryManifestPoller::new(uplink_config).await?)
+                    Arc::new(
+                        PersistedQueryManifestPoller::new(
+                            configuration
+                                .uplink
+                                .as_ref()
+                                .expect("uplink config was checked above, qed"),
```

Comment:
Is this as tight as it was before?

Reply:
This was how @BrynCooke refactored it - I had introduced _two_ new structs, `UplinkConfig` and `PartialUplinkConfig` and converted between them with error states. @BrynCooke's refactor moves the "require `APOLLO_KEY` and `APOLLO_GRAPH_REF`" step earlier in the pipeline, and so by the time you get here, you're guaranteed to have the configuration. I still don't love having an `expect` but 🤷🏼

Reply:
fair - i updated it to remove the `expect` and put it back in the top function